### PR TITLE
enable ci

### PR DESCRIPTION
**Pull request by tksugimoto**
## 参考
- [akka-entity-replication/ci.yml at v1.0.0 · lerna-stack/akka-entity-replication](https://github.com/lerna-stack/akka-entity-replication/blob/v1.0.0/.github/workflows/ci.yml)
- [lerna-sample-payment-app/ci.yml at v1.1.0 · lerna-stack/lerna-sample-payment-app](https://github.com/lerna-stack/lerna-sample-payment-app/blob/v1.1.0/.github/workflows/ci.yml)

## 確認済み
- compile 時 warn で failed
  - https://github.com/lerna-stack/lerna-app-library/pull/1#discussion_r611299159
- test ERROR で failed
  - https://github.com/lerna-stack/lerna-app-library/pull/1#discussion_r611302214

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+      - name: Run tests
+        continue-on-error: true # results are reported by action-junit-report
+        run: sbt coverage test
+
+      - name: Publish test report
+        uses: mikepenz/action-junit-report@v2
+        with:
+          check_name: ScalaTest Report
+          report_paths: 'target/**test-reports/TEST-*.xml'
```
Comment (tksugimoto) [inline .github/workflows/ci.yml:51]: [enable ci by tksugimoto · Pull Request #1 · lerna-stack/lerna-app-library](https://github.com/lerna-stack/lerna-app-library/pull/1/checks?check_run_id=2304671599)

> GitHub Actions / ScalaTest Report  
> failed 2021/04/09(金) 18:31:05 (23 minutes ago) in 0s  
> No test results found!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  SBT_OPTS: -DSBT_SCALAC_STRICT_WARNINGS=true
```
Comment (tksugimoto) [inline .github/workflows/ci.yml:12]: [enable ci by tksugimoto · Pull Request #1 · lerna-stack/lerna-app-library](https://github.com/lerna-stack/lerna-app-library/pull/1/checks?check_run_id=2304833961)

> [warn]  [E1] lerna-http/src/main/scala/lerna/http/directives/RequestLogDirective.scala  
> [warn]       Unused import  
> [warn]       L4: import akka.http.scaladsl.server.{ AuthenticationFailedRejection, Directive0, RouteResult }  
> [warn]                                              ^  
> [warn] lerna-http/src/main/scala/lerna/http/directives/RequestLogDirective.scala: L4 [E1]  

未使用importチェックできてない

Comment (tksugimoto) [reply .github/workflows/ci.yml:12]: - 環境変数は `sys.env`
- `-D` プロパティは `sys.props`

Comment (tksugimoto) [reply .github/workflows/ci.yml]:
```diff
 env:
-  SBT_OPTS: -DSBT_SCALAC_STRICT_WARNINGS=true
+  SBT_SCALAC_STRICT_WARNINGS: true

 jobs:
```

[enable ci by tksugimoto · Pull Request #1 · lerna-stack/lerna-app-library](https://github.com/lerna-stack/lerna-app-library/pull/1/checks?check_run_id=2320170510#step:6:33)

> Error:  [E1] lerna-http/src/main/scala/lerna/http/directives/RequestLogDirective.scala  
> Error:       Unused import  
> Error:       L4: import akka.http.scaladsl.server.{ AuthenticationFailedRejection, Directive0, RouteResult }  
> Error:                                              ^  
> Error:  lerna-http/src/main/scala/lerna/http/directives/RequestLogDirective.scala: L4 [E1]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tksugimoto) [reply .github/workflows/ci.yml]: ef0ad09
```diff
@@ -48,7 +48,7 @@ jobs:
         uses: mikepenz/action-junit-report@v2
         with:
           check_name: ScalaTest Report
-          report_paths: 'target/**test-reports/TEST-*.xml'
+          report_paths: 'lerna-*/target/test-reports/TEST-*.xml'
           github_token: ${{ secrets.GITHUB_TOKEN }}
```

ちゃんと test 落ちでレポートがupされて、failedが存在したらCI failed

(docker で cassandraを動かしていないnode落ちた)

[enable ci by tksugimoto · Pull Request #1 · lerna-stack/lerna-app-library](https://github.com/lerna-stack/lerna-app-library/pull/1/checks?check_run_id=2320221403)

> CassandraSequenceFactorySpec.(It is not a test it is a sbt.testing.SuiteSelector)  
> com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /127.0.0.1:9042 (com.datastax.driver.core.exceptions.TransportException: [/127.0.0.1:9042] Cannot connect))  
> Raw output

Comment (tksugimoto) [reply .github/workflows/ci.yml]: > docker で cassandraを動かしていないため落ちた

`docker-compose up -d cassandra` 追加

```diff
      - name: Compile
        run: sbt clean compile test:compile

      # https://github.com/marketplace/actions/docker-layer-caching
      - name: Pull the latest docker image
        run: docker-compose pull cassandra
      - name: Docker layer cache
        uses: satackey/action-docker-layer-caching@v0.0.11
        continue-on-error: true

      - name: Build the docker-compose stack
        run: docker-compose up -d cassandra

      - name: Run tests
        continue-on-error: true # results are reported by action-junit-report
        run: sbt coverage test
```

[enable ci by tksugimoto · Pull Request #1 · lerna-stack/lerna-app-library](https://github.com/lerna-stack/lerna-app-library/pull/1/checks?check_run_id=2320366537)

> 158 tests run, 2 skipped, 0 failed.

**Review by xirc — APPROVED** [review]
LGTM 👍

**Pull request merged**